src/components/button.tsx

```diff
@@ -37,17 +37,29 @@ export const Button: FC<Props> = ({
     "min-h-52": size === "large",
     "min-h-36 lg:min-h-52": size === "responsive",
   })
-  const classes = classnames("rounded border", {
-    "bg-teal hover:bg-teal-dark focus:bg-teal border-teal": style === "teal",
-    "bg-white text-teal hover:opacity-60 border-teal": style === "teal-border",
-    "bg-salmon border-salmon hover:bg-salmon-dark focus:bg-salmon":
-      style === "salmon",
-    "cursor-not-allowed ": disabled,
-    "bg-grey-1 border-grey-1 hover:bg-grey-1 text-grey-4":
-      disabled && style !== "teal-border",
-    "text-grey-3 border-grey-3 hover:opacity-100":
-      disabled && style === "teal-border",
-  })
+
+  const buttonStyles = {
+    teal: "bg-teal hover:bg-teal-dark focus:bg-teal border-teal",
+    "teal-border": "bg-white text-teal hover:opacity-60 border-teal",
+    white: "bg-none text-white hover:opacity-60 border-white",
+    salmon: "bg-salmon border-salmon hover:bg-salmon-dark focus:bg-salmon",
+  }
+
+  const disabledStyles = {
+    teal: "bg-grey-1 border-grey-1 hover:bg-grey-1 text-grey-4",
+    "teal-border": "text-grey-3 border-grey-3 hover:opacity-100",
+    white: "text-grey-3 border-grey-3 hover:opacity-100",
```

Comment:
Is it possible to make a clean up here instead of repeating the same styles?

Reply:
I duplicated on purpose because personally I think it's more visible what happens when we duplicate it. Before it was already quite complex with a lot of conditions